libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -405,6 +405,16 @@ void AP_MotorsMatrix::output_armed_stabilizing()
     check_for_failed_motor(throttle_thrust_best_plus_adj);
 }
 
+// return throttle out for motor motor_num, returns true if value is valid false otherwise
+bool AP_MotorsMatrix::get_throttle_out(uint8_t motor_num, float& thr_out) const
+{
+    if (motor_num >= AP_MOTORS_MAX_NUM_MOTORS || !motor_enabled[motor_num]) {
+        return false;
+    }
+    thr_out = _thrust_rpyt_out[motor_num] / thr_lin.get_compensation_gain();
```

Comment:
I think you will find `_thrust_rpyt_out` is wrong when motors are not in throttle unlimited. You should be able to use `_actuator` (which also obeys slew limiting) there is a `thr_lin.actuator_to_thrust` method to remove the expo to get back to the original value. Because that is a MotorsMulticopter level variable you should be able to move this up and have it work for tricopter ect too.

Reply:
Also you will need a `abs` too in that case, the 6dof backed can output negative values.

Reply:
Where do you think it is changed when throttle limited?

Reply:
Its simply that `_thrust_rpyt_out` is not output to the motors but `output_armed_stabilizing` is still run. So `_thrust_rpyt_out` could have some none zero stuff in it. It depends how good a job were doing keeping all the controllers zeroed (copter is probably better than plane).

Independent of that if you have some motor slew limits set then `_thrust_rpyt_out` by-passes the that in throttle unlimited. 

It would also be nice if the filters tacked up with spool up. Of course the lower harmonics would have zero attenuation thanks to the recent re-work, but the higher harmonics would be active. This is what would happen with RPM feedback, I presume the goal is to match that as closely as possible.

Reply:
I agree with you that there are parts of this that slew rate factors that are not included however the complexity of removing slew rate and linearisation doesn't justify the gains in my opinion.

Reply:
There are several benefits to using `_actuator`, I do think its worth doing. These are the ones I can think of at the moment:

- Slew limits, spool up and down, E-Stop and interlock all work correctly. 
- Works for all `AP_MotorsMulticopter` derived classes, not just `AP_MotorsMatrix`. Tricopter, coax, single, tailsitter.
- Works in forward flight for Quadplanes using the `output_motor_mask` method.

Reply:
The log from @andyp1per highlights the first point. ESC driven in green, throttle driven in red. The PWM output in blue. 

![image](https://github.com/ArduPilot/ardupilot/assets/33176108/66e69a32-572a-4a2c-a4aa-6d8eb83a87de)

We see the throttle based in red go to zero while the vehicle is in ground idle. Where as the PWM output derived from the `_actuator` value tracks up and down with the true RPM. `_actuator` includes spin arm/min/max so can track in those cases. Although we might have to have a think about best remove the linearisation outside the spin range.

Reply:
Ok @IamPete1 's change checked by @lthall and incorporated